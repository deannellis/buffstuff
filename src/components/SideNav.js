import React, { Component } from "react";
import { Link } from "react-router-dom";
import { MenuContext } from "../context/menu-context";
import { DumbbellIcon, DashboardIcon, GraphIcon } from "../utils/icons";

export const navList = [
  {
    label: "Dashboard",
    icon: "dashboard",
    path: "/home"
  },
  {
    label: "Lifts",
    icon: "graph",
    path: "/lifts"
  },
  {
    label: "Workouts",
    icon: "dumbbell",
    path: "/workouts"
  }
];

class SideNav extends Component {
  render() {
    let { menuIsOpen } = this.context;
    const getIcon = name => {
      switch (name) {
        case "dumbbell":
          return <DumbbellIcon size={24} />;
        case "dashboard":
          return <DashboardIcon size={24} />;
        case "graph":
          return <GraphIcon size={24} />;
        default:
          return <></>;
      }
    };

    return (
      <div className="side-nav__container">
        <div
          className={menuIsOpen ? "side-nav side-nav--opened" : "side-nav"}
          data-testid={"side-nav"}
        >
          {navList.map(item => (
            <Link to={item.path} key={item.path}>
              <div
                className={`side-nav__nav-item ${
                  this.props.path == item.path
                    ? "side-nav__nav-item--selected"
                    : ""
                }`}
              >
                {getIcon(item.icon)}
                <span>{item.label}</span>
              </div>
            </Link>
          ))}
        </div>
      </div>
    );
  }
}
SideNav.contextType = MenuContext;

export default SideNav;
