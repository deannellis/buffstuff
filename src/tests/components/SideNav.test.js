import React from 'react';
import { shallow } from 'enzyme';

import SideNav from '../../components/SideNav';

test('should render SideNav', () => {
    const wrapper = shallow(<SideNav />);
    expect(wrapper).toMatchSnapshot();
});

// test('should render SideNav', () => {});
