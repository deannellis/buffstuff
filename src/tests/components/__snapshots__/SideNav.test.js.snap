// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render SideNav 1`] = `
<div
  className="side-nav__container"
>
  <div
    className="side-nav"
  >
    <Link
      key="/home"
      to="/home"
    >
      <div
        className="side-nav__nav-item "
      >
        <DashboardIcon
          size={24}
        />
        <span>
          Dashboard
        </span>
      </div>
    </Link>
    <Link
      key="/lifts"
      to="/lifts"
    >
      <div
        className="side-nav__nav-item "
      >
        <GraphIcon
          size={24}
        />
        <span>
          Lifts
        </span>
      </div>
    </Link>
    <Link
      key="/workouts"
      to="/workouts"
    >
      <div
        className="side-nav__nav-item "
      >
        <DumbbellIcon
          size={24}
        />
        <span>
          Workouts
        </span>
      </div>
    </Link>
  </div>
</div>
`;
